CommandLine/libEGM/action.cpp

```diff
@@ -148,16 +149,19 @@ std::string Actions2Code(const std::vector< buffers::resources::Action >& action
           code += ')';
         }
 
-        if (action.relative())
+        if (action.use_relative())
           code += action.is_question() ? ");" : "\n}";
         if (action.is_question()) {
           code += "\nif (__if__)";
         }
+        code += "\n";
+
+        if (action.who_name() != "self" && !action.is_question())

```

Comment:
This may work for GMX, but I don't know that it works for GMK because GMK only contains integers. For GMK it will be a `-1` instead of self. Would probably make sense to fix it in the GMK reader, in which case you can bring back that code you had in the lambda but invert the mapping it does.

Reply:
Wouldn't it be easier to have something like `(action.who_name() != "self" || action.id() != -1)` which will work for both cases?

Reply:
It largely doesn't matter but I'm not sure this is the only place to handle it, that's why I think it's better to handle at the source.